From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Tue, 4 Aug 2026 10:57:17 +0000
Subject: rockchip: rk3506: Remove unneeded syscon driver

Modern Rockchip device drivers use syscon_regmap_lookup_by_phandle()
to get a regmap or use a constant to get the fixed base address.

Remove the unneeded RK3506 syscon driver to discourage new uses of
syscon_get_regmap_by_driver_data() or syscon_get_first_range().

The generic_syscon driver will instead be used for the compatible
previously matched by the now removed driver.

Signed-off-by: Austin Lane <vidplace7@gmail.com>
---
 arch/arm/mach-rockchip/rk3506/Makefile        |  1 -
 arch/arm/mach-rockchip/rk3506/syscon_rk3506.c | 19 ----------
 2 files changed, 20 deletions(-)

diff --git a/arch/arm/mach-rockchip/rk3506/Makefile b/arch/arm/mach-rockchip/rk3506/Makefile
index 111111111111..222222222222 100644
--- a/arch/arm/mach-rockchip/rk3506/Makefile
+++ b/arch/arm/mach-rockchip/rk3506/Makefile
@@ -2,4 +2,3 @@
 
 obj-y += rk3506.o
 obj-y += clk_rk3506.o
-obj-y += syscon_rk3506.o
diff --git a/arch/arm/mach-rockchip/rk3506/syscon_rk3506.c b/arch/arm/mach-rockchip/rk3506/syscon_rk3506.c
deleted file mode 100644
index 111111111111..222222222222
--- a/arch/arm/mach-rockchip/rk3506/syscon_rk3506.c
+++ /dev/null
@@ -1,19 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
-// Copyright Contributors to the U-Boot project.
-
-#include <dm.h>
-#include <asm/arch-rockchip/clock.h>
-
-static const struct udevice_id rk3506_syscon_ids[] = {
-	{ .compatible = "rockchip,rk3506-grf", .data = ROCKCHIP_SYSCON_GRF },
-	{ }
-};
-
-U_BOOT_DRIVER(rockchip_rk3506_syscon) = {
-	.name = "rockchip_rk3506_syscon",
-	.id = UCLASS_SYSCON,
-	.of_match = rk3506_syscon_ids,
-#if CONFIG_IS_ENABLED(OF_REAL)
-	.bind = dm_scan_fdt_dev,
-#endif
-};
-- 
Armbian

